easybuild/easyblocks/i/impi.py

```diff
@@ -149,6 +154,41 @@ def post_processing_step(self):
                     if os.path.exists(wrapper_path):
                         apply_regex_substitutions(wrapper_path, regex_subs)
 
+        # Rebuild Fortran 2008 bindings if requested.
+        # For more information, see:
+        # https://community.intel.com/t5/Intel-MPI-Library/MPI-f08-with-polymorphic-argument-CLASS/m-p/1590421
+        if self.cfg['rebuild_f08_bindings']:
+            # Check if the binding tarball exists within the installation
+            bindings_path = os.path.join(self.installdir, 'mpi', 'latest', 'opt', 'mpi', 'binding')
+            bindings_tarball = os.path.join(bindings_path, 'intel-mpi-binding-kit.tar.gz')
+            if not os.path.exists(bindings_tarball):
+                raise EasyBuildError(
+                    f"Requested to rebuild Fortran 2008 bindings, but the bindings tarball in {bindings_tarball} "
+                    f"does not exist.")
+            self.log.info(f"Found bindings tarball at {bindings_tarball}. Rebuilding Fortran 2008 bindings.")
+            # Determine compilers to build with
+            if self.toolchain.is_system_toolchain():
+                f90_compiler = 'gfortran'
+            else:
+                f90_compiler = self.toolchain.COMPILER_F90
+            # Extract the tarball
+            build_dir = tempfile.mkdtemp()
+            run_shell_cmd(f"tar -xzf {bindings_tarball} -C {build_dir}")
+            # Build the bindings
+            change_dir(os.path.join(build_dir, 'f08'))
+            run_shell_cmd(f"make MPI_INST={self.installdir}/mpi/latest F90={f90_compiler} NAME={f90_compiler}")
+            change_dir(os.path.join(get_cwd(), 'include', f'{f90_compiler}'))
+            initial_module_files = glob.glob(f"{self.installdir}/mpi/latest/include/mpi/*.mod")
+            written_module_files = glob.glob("*.mod")
+            # Preserve the initial module files for people to use
+            mkdir(f"{self.installdir}/mpi/latest/include/mpi/back", parents=True)
```

Comment:
I guess the `back` is short for backup?

Makes sense to introduce a local variable for this path, since it must be identical to what's used below.

I would also introduce a dedicated method for this whole code block.

Done in https://github.com/Thyre/easybuild-easyblocks/pull/1